apps/backend/src/app/auctions/auctions.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateAuctionDto } from './dto/create-auction.dto';
+import { UpdateAuctionDto } from './dto/update-auction.dto';
+import { Auction } from './entities/auction.entity';
+
+@Injectable()
+export class AuctionsService {
+  constructor(
+    @InjectRepository(Auction)
+    private readonly auctionsRepository: Repository<Auction>
+  ) {}
+
+  create(createAuctionDto: CreateAuctionDto) {
+    const auction = this.auctionsRepository.create(createAuctionDto);
+
+    return this.auctionsRepository.save(auction);
+  }
+
+  findAll() {
+    return this.auctionsRepository.find();
+  }
+
+  async findOne(id: number) {
+    const auction = await this.auctionsRepository.findOneBy({ id });
+
+    if (!auction) {
+      throw new NotFoundException(`Auction #${id} not found`);
```

Comment:
I'm not sure we should throw http exceptions from service. This is how things happen in nestjs? 😎

Reply:
Alright, took it from the tutorial, will think about it 😄